src/Integration.UnitTests/Helpers/AbsoluteFilePathLocatorTests.cs

```diff
@@ -36,20 +36,31 @@ public class AbsoluteFilePathLocatorTests
     {
         [TestMethod]
         public void MefCtor_CheckIsExported()
+            => MefTestHelpers.CheckTypeCanBeImported<AbsoluteFilePathLocator, IAbsoluteFilePathLocator>(
+                MefTestHelpers.CreateExport<IVsUIServiceOperation>());
+
+        [TestMethod]
+        public void MefCtor_CheckIsSingleton()
+            => MefTestHelpers.CheckIsSingletonMefComponent<AbsoluteFilePathLocator>();
+
+        [TestMethod]
+        public void MefCtor_DoesNotCallAnyServices()
         {
-            var serviceProvider = new Mock<IServiceProvider>();
-            serviceProvider.Setup(x => x.GetService(typeof(SVsUIShellOpenDocument))).Returns(Mock.Of<IVsUIShellOpenDocument>());
+            var serviceOp = new Mock<IVsUIServiceOperation>();
 
-            MefTestHelpers.CheckTypeCanBeImported<AbsoluteFilePathLocator, IAbsoluteFilePathLocator>(
-                MefTestHelpers.CreateExport<SVsServiceProvider>(serviceProvider.Object));
+            _ = CreateTestSubject(serviceOp.Object);
+
+            // The MEF constructor should be free-threaded, which it will be if
+            // it doesn't make any external calls.
+            serviceOp.Invocations.Should().BeEmpty();
         }
 
         [TestMethod]
         public void Locate_RelativePathIsNull_ArgumentNullException()
         {
             var vsUiShell = new Mock<IVsUIShellOpenDocument>();
             
-            var testSubject = new AbsoluteFilePathLocator(CreateServiceProvider(vsUiShell.Object));
```

Comment:
Miscellaneous: added a `CreateTestSubject` method, even though the class constructor only has one parameter.